classes/output/login.php

```diff
@@ -49,51 +53,84 @@ public static function output_login_options($wantsurl) {
             }
             $authplugin = get_auth_plugin($authname);
             $potentialidps = $authplugin->loginpage_idp_list($wantsurl);
-
             if (!empty($potentialidps)) {
                 foreach ($potentialidps as $idp) {
-                    $output .= \html_writer::start_tag('div', array('class' => 'idp-login container-fluid '. $authname));
-                    $name = '';
+                    $idpcontext = [
+                        'auth' => $authname,
+                        'name' => s($idp['name']),
+                        'url' => $idp['url']
+                    ];
                     if (!empty($idp['iconurl'])) {
-                        $name .= '<img src="' . s($idp['iconurl']) . '" width="24" height="24" class="mr-1"/>';
+                        $idpcontext['iconurl'] = s($idp['iconurl']);
                     }
-                    $name .= s($idp['name']);
-                    $attributes = ['class' => 'btn btn-secondary btn-block', 'title' => s($idp['name'])];
-                    $output .= \html_writer::link($idp['url'], $name, $attributes);
-
-                    $output .= \html_writer::end_tag('div');
-                    $idploginpath = $idp['url'];
+                    $idplist[] = (object) $idpcontext;
                     $count++;
+                    $idploginpath = $idp['url'];
```

Comment:
Could we scrap this variable and then on line 77 just use `current($idplist)['url']` ? 

To me having a variable set and overwritten by the for loop is difficult to read